### PR TITLE
[automation-api/nodejs] - Support recovery workflow

### PR DESCRIPTION
**Pull request by komalali**
Resolves: https://github.com/pulumi/pulumi/issues/5531

### CODE REVIEW

File: sdk/nodejs/x/automation/workspace.ts [hunk 1]
```diff
@@ -202,6 +211,14 @@ export interface StackSummary {
     url?: string;
 }
 
+/**
+ * Deployment encapsulates the state of a stack deployment.
+ */
+export interface Deployment {
+    version: number;
+    deployment: object;
```
Comment (komalali) [inline sdk/nodejs/x/automation/workspace.ts:219]: Is there a better type for this? In [Go](https://github.com/pulumi/pulumi/blob/cedab0003bda3b4cbb767d1a8875a620c34ba9da/sdk/go/common/apitype/core.go#L159-L167) we've left this unmarshalled but it's not possible to partially parse a json object in Typescript as far as I'm aware.

Comment (EvanBoyle) [reply sdk/nodejs/x/automation/workspace.ts:219]: The type of deployment ends up being one of `DeploymentV1/2/3`: https://github.com/pulumi/pulumi/blob/48f906e0263335ad5fc922f6836b75c5f27da5eb/sdk/go/common/apitype/core.go#L108

That API in go is designed for backwards/forwards compat. SerDe and typing in typescript are far more flexible than in Go, so we may be able to improve the experience here with union types. If you're going to stick with the untyped route that requires the user to do type assertions, `any` may be more idiomatic.

**Review by EvanBoyle — APPROVED** [review]

File: sdk/nodejs/x/automation/workspace.ts (hunk 1 above)
Comment (komalali) [reply sdk/nodejs/x/automation/workspace.ts:219]: Ah perfect, thanks for the pointers. I'll fix this up with a union of the deployment versions.

File: sdk/nodejs/x/automation/localWorkspace.ts [hunk 2]
```diff
@@ -493,6 +493,28 @@ export class LocalWorkspace implements Workspace {
             return value;
         });
     }
+    /**
+     * exportStack exports the deployment state of the stack.
+     * This can be combined with Workspace.importStack to edit a stack's state (such as recovery from failed deployments).
```
[the remark this answers is not in the document]
Comment (komalali) [reply sdk/nodejs/x/automation/localWorkspace.ts:498]: Can you clarify what you mean by `interfaces` when you refer to other areas where this needs to be fixed?

[the remark this answers is not in the document]
Comment (EvanBoyle) [reply sdk/nodejs/x/automation/localWorkspace.ts:498]: I was referring to methods in the `workspace` interface. Documenting params of the methods that were added similar to how the existing methods are documented: https://github.com/pulumi/pulumi/pull/6038/files/d434da0774e7ced3a6ca69907a84779ab7e5f248#diff-5d90703c04c98bf207aef5a82697fa29f4655353f9f02a9ed83902abce9c6b2aL180-L185

File: sdk/nodejs/x/automation/workspace.ts (hunk 1 above)
Comment (komalali) [reply sdk/nodejs/x/automation/workspace.ts:219]: Okay I'm realizing this is a bottomless hole so I'm gonna stick with `any` and leave a note to make it better in the future.

**Pull request merged**